src/gstools/variogram/variogram.py

```diff
@@ -471,7 +559,10 @@ def vario_estimate_axis(
         if missing:
             field.mask = np.logical_or(field.mask, missing_mask)
         mask = np.ma.getmaskarray(field)
-        if not config.USE_RUST:
+        if (
+            not config.USE_GSTOOLS_CORE
+            and config._GSTOOLS_CORE_AVAIL  # pylint: disable=W0212
+        ):  # pylint: disable=W0212
             mask = np.asarray(mask, dtype=np.int32)
     else:
         field = np.atleast_1d(np.asarray(field, dtype=np.double))
```

Comment:
Ahh, that is a compatibility issue. Now I get it. So we use bool in both cases now? Can this be removed then?!

Reply:
Indeed, good eye ;-)

Reply:
Since we now use the correct type for numpy bools, this conversion also should not happen.

Reply:
Yeah, in the newest commits, I dropped those lines. I see that that was not clear from my previous message ;-)